### DISCUSSION TITLE
With the new app router, can you still render components on the server, and then also make them client side?

### DISCUSSION THREAD

**Discussion by avisra** [vercel/next.js#54886] [App Router] [answered]
Let's take a product listing page for example. I'd prefer the products, without any filters selected, by loaded from the server... but then on page load, allow the products to be filtered / updated dynamically when the user applies categories or search criteria. This is for SEO purposes. This was achievable with the pages router - curious if this same concept is achievable with the app router.

Thanks

**Comment by Dragate** [comment, marked as answer]
[Yes](https://nextjs.org/docs/app/building-your-application/rendering/client-components#how-are-client-components-rendered)
